tests/base/src/test/java/org/wildfly/security/http/HttpAuthenticatorTest.java

```diff
@@ -255,5 +256,24 @@ public void testUsingSecurityProviderServerMechanismWithDigestMD5() throws Excep
 
         authenticateWithDigestMD5();
     }
-
+    @Test
+    public void testBearerAuthenticationMechanism() throws Exception{
+        HttpServerAuthenticationMechanism mechanism = bearerFactory.createAuthenticationMechanism(BEARER_TOKEN,Collections.emptyMap(),getCallbackHandler("remoteUser","testrealm@host.com",null,"random"));
+
+        //Test no authentication in progress
+        TestingHttpServerRequest request1 = new TestingHttpServerRequest(new String[]{});
+        mechanism.evaluateRequest(request1);
```

Comment:
@keshav-redhat  IMO here if you would configure `request.setRemoteUser("remoteUser");` you would be able to remove the hard coded values from the mechanism. But the remote user should not have to be configured for bearer test. It should only be needed for external mechanism. 

So on the place where you get an exception when remote user is not provided, you should rather modify the test class to not expect the external user to be configured when bearer mechanism is used

Reply:
Thank you for reviewing the PR @Skyllarr .I have updated the check in `AbstractBaseHttpTest.java`.Please have a look on this.